src/Type/Constant/ConstantIntegerType.php

```diff
@@ -88,6 +91,40 @@ public function generalize(GeneralizePrecision $precision): Type
 		return new IntegerType();
 	}
 
+	public function exponentiate(Type $exponent): Type
+	{
+		if ($exponent instanceof IntegerRangeType) {
```

Comment:
What about `ConstantIntegerType ** "unions of integer ranges"`? Where is it handled?